satellite/satellitedb/mutex.go

```diff
@@ -0,0 +1,267 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package satellitedb
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"storj.io/storj/pkg/accounting"
+	"storj.io/storj/pkg/bwagreement"
+	"storj.io/storj/pkg/datarepair/irreparable"
+	"storj.io/storj/pkg/datarepair/queue"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/statdb"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/storage"
+)
+
+// Mutex wraps DB in a mutex for non-concurrent databases
+type Mutex struct {
+	mu sync.Mutex
+	db *DB
+}
+
+// NewMutex returns a mutex around DB
+func NewMutex(db *DB) *Mutex {
+	return &Mutex{db: db}
+}
+
+func (db *Mutex) locked() func() {
+	db.mu.Lock()
+	return db.mu.Unlock
+}
+
+// BandwidthAgreement is a getter for bandwidth agreement repository
+func (db *Mutex) BandwidthAgreement() bwagreement.DB {
+	return &muBandwidthAgreement{mu: db, db: db.db.BandwidthAgreement()}
+}
+
+// // PointerDB is a getter for PointerDB repository
```

Comment:
Is there a reason this is commented out? Should we remove this?

Reply:
PointerDB isn't yet implemented for master database.